lib/conjur/base.rb

```diff
@@ -351,8 +351,12 @@ def mtime
       end
 
       def refresh_token
+        # There's a race condition here in which the file could be updated
+        # after we read the mtime but before we read the file contents. So to be
+        # conservative, use the oldest possible mtime.
+        mtime = self.mtime
```

Comment:
That's exactly why I was using mtime from `f`! It's guaranteed to correspond to what's actually read. There was no race here (in your original code, there was).